ibc/src/component/connection.rs

```diff
@@ -55,7 +55,26 @@ impl Component for ConnectionComponent {
     #[instrument(name = "ibc_connection", skip(tx))]
     fn check_tx_stateless(tx: &Transaction) -> Result<()> {
         for ibc_action in tx.ibc_actions() {
-            validate_ibc_action_stateless(ibc_action)?;
+            match &ibc_action.action {
+                Some(ConnectionOpenInit(msg)) => {
+                    use stateless::connection_open_init::*;
+                    let msg = MsgConnectionOpenInit::try_from(msg.clone())?;
+
+                    version_is_supported(&msg)?;
+                }
+                Some(ConnectionOpenTry(msg)) => {
+                    let _ = MsgConnectionOpenTry::try_from(msg.clone())?;
+                }
+                Some(ConnectionOpenAck(msg)) => {
+                    let _ = MsgConnectionOpenAck::try_from(msg.clone())?;
+                }
+
+                Some(ConnectionOpenConfirm(msg)) => {
+                    let _ = MsgConnectionOpenConfirm::try_from(msg.clone())?;
+                }
```

Comment:
Are these supposed to have more checks added later? If so, could we have a TODO comment, so that when reading the code we can assume that either the checks listed there are all of the checks, or else there's an explicit marker that some are left to be implemented?

Reply:
I wasn't sure if there was any stateless validation that could be pulled out of the stateful validation, but there is for OpenTry and OpenAck (that client state, client proof, and consensus proof are not None), so I pushed a commit to pull those validations into the stateless mod and document that OpenConfirm has no additional stateless validation